### PR TITLE
[WIP] Bug Fix: `azurerm_application_gateway` - updatable gateway subnet

### PR DESCRIPTION
**Pull request by mbfrahry**
Tests passing: 

image

### CODE REVIEW

File: azurerm/resource_arm_application_gateway.go [hunk 1]
```diff
@@ -1175,6 +1176,17 @@ func resourceArmApplicationGatewayCreateUpdate(d *schema.ResourceData, meta inte
 		gateway.ApplicationGatewayPropertiesFormat.WebApplicationFirewallConfiguration = expandApplicationGatewayWafConfig(d)
 	}
 
+	if stopApplicationGateway {
+		future, err := client.Stop(ctx, resGroup, name)
+		if err != nil {
```
Comment (katbyte) [inline azurerm/resource_arm_application_gateway.go]: What will happen here if it's already stopped? 🤔

**Review by katbyte — APPROVED** [review]
Aside from one comment LGTM 👍

File: azurerm/resource_arm_application_gateway.go (hunk 1 above)
Comment (mbfrahry) [reply azurerm/resource_arm_application_gateway.go]: Just checked by adding another stop section and it passed just fine. I think we're good!

**Comment by mbfrahry** [comment]
Addressing one of the points of #1576

**Pull request merged**